examples/search_examples/public/search/app.tsx

```diff
@@ -304,14 +317,30 @@ export const SearchExamplesApp = ({
 
       setRequest(searchSource.getSearchRequestBody());
       const abortController = new AbortController();
+      const inspector: IInspectorInfo = {
+        adapter: new RequestAdapter({ handleWarnings: data.search.showWarnings }),
```

Comment:
should we use `data.search.showWarnings` as the default implementation? it seems like in most cases this should be the fallback. The way I plan to use this feature:
* Disable default handling on the expression
* Call handleWarnings on the adapter, passing in my own warning handling logic as a callback
* If my own logic is able to handle all of the warnings, everything's good, however if there are unexpected errors I want to just delegate to the default handler (right now I would have to set that default handler when creating the adapter instance)

What do you think? Am I missing something here?

Reply:
Hi @flash1293 You are not missing anything - this PR should tie things together the way you expected. I saw this as something that needed to be changed in my PR: https://github.com/elastic/kibana/pull/138342/commits/27957111f35d2d37885bffa36351297fad0a2ee7 

> Call handleWarnings on the adapter, passing in my own warning handling logic as a callback

This is possible in the current state of the code. When calling the `handleWarnings` method, you will need to pass in an `inspector` object with a request adapter that has collected the requests for your query. Hope that will not be a problem :).